app/viewModels/PaginationViewModel.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewModels
+
+import controllers.testOnly.routes
+import play.api.libs.json.{JsObject, Json}
+
+object PaginationViewModel {
+
+  def apply(
+    totalNumberOfMovements: Int,
+    currentPage: Int,
+    numberOfMovementsPerPage: Int
+  ): JsObject = {
+
+    val numberOfPagesFloat = totalNumberOfMovements.toFloat / numberOfMovementsPerPage
+
+    val totalNumberOfPages = if (numberOfPagesFloat.isWhole) {
+      numberOfPagesFloat.toInt
+    } else {
+      numberOfPagesFloat.toInt + 1
+    }
+
+    val validateCurrentPage = if (currentPage < 1 | currentPage > totalNumberOfPages) 1 else currentPage
+
+    def buildItem(pageNumber: Int, dottedLeft: Boolean, dottedRight: Boolean): JsObject = Json.obj(
+      "pageNumber"  -> pageNumber,
+      "href"        -> s"${routes.ViewAllArrivalsController.onPageLoad(Some(pageNumber)).url}", // TODO move this to param
```

Comment:
This needs to be moved to a param, but could be done in the destination pagination branch